cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -42,7 +46,7 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(AbstractProgramRuntimeService.class);
 
-  private final Table<ProgramType, RunId, RuntimeInfo> runtimeInfos;
+  protected final Table<ProgramType, RunId, RuntimeInfo> runtimeInfos;
```

Comment:
This is dangerous. I don't like exposing raw data structure directly to children classes.